es6/animations.js

```diff
@@ -188,74 +184,116 @@ export const AnimateCSS = ({name, path, start_time, end_time, duration=1000, cur
     if (start_state === undefined) start_state = {top: 0, left: 0}
+
+    if (exactlyOneIsUndefined(delta_state, end_state)) {
+        [delta_state,
+            end_state] = computeTheOther(start_state, delta_state, end_state)
+    }
+
     path = `${path}/style/transform/translate`
     const type = 'TRANSLATE'
 
-    const animation = {type, path, start_time, end_time, duration, start_state, end_state, amt, curve, unit}
+    const animation = {type, path, start_time, end_time, duration,
+                       start_state, end_state, delta_state, curve, unit}
 
-    const left_tick = unit_tick(animation, 'left', 0)  //  TODO: change left => /left to keep state selectors consistent
-    const top_tick =  unit_tick(animation, 'top', 0)
+    const left_tick = tick_func({
+        duration,
+        curve,
+        unit,
+        start_state: start_state['left'],
+        delta_state: delta_state['left'],
+    })
+    const top_tick = tick_func({
+        duration,
+        curve,
+        unit,
+        start_state: start_state['top'],
+        delta_state: delta_state['top'],
+    })
 
-    animation.tick = (delta) => ({
-        left: left_tick(delta),
-        top: top_tick(delta),
+    animation.tick = (time_elapsed) => ({
+        left: left_tick(time_elapsed),
+        top: top_tick(time_elapsed),
     })
     return Animate(animation)
 }
 
-export const TranslateTo = ({path, start_time, end_time, duration=1000, start_state, end_state, amt, curve='linear', unit='px'}) => {
-    let anims = []
-    const has_left = (start_state || end_state || amt).left !== undefined
-    const has_top = (start_state || end_state || amt).top !== undefined
-    if (has_left) {
-        anims = [
-            KeyedAnimation({
-                type: 'TRANSLATE_TO_LEFT',
-                path: `${path}/style`,
-                key: 'left',
-                start_time, end_time, duration,
-                start_state, end_state, amt,
-                curve, unit,
-            })
-        ]
+export const Style = ({
+        path, start_time, end_time, duration=1000, start_state, end_state,
+        delta_state, curve='linear', unit='px'}) => {
+    if (start_time === undefined) start_time = (new Date).getTime()
+
+    if (exactlyOneIsUndefined(duration, end_time)) {
+        [duration,
+            end_time] = computeTheOther(start_time, duration, end_time)
     }
-    if (has_top) {
-        anims = [
-            ...anims,
-            KeyedAnimation({
-                type: 'TRANSLATE_TO_TOP',
-                path: `${path}/style`,
-                key: 'top',
-                start_time, end_time, duration,
-                start_state, end_state, amt,
-                curve, unit,
-            })
-        ]
+
+
+    if (exactlyOneIsUndefined(delta_state, end_state)) {
+        [delta_state,
+            end_state] = computeTheOther(start_state, delta_state, end_state)
+    }
+
+    const tick_funcs = {}
+    Object.keys(start_state).forEach((key) => {
```

Comment:
why not use `mapObj`? It's built for exactly this.  Alternatively `reduce` or `for (let key of start_state)` plz.